app/src/main/java/org/sagebionetworks/research/mpower/history/MpHistoryItemManager.kt

```diff
@@ -95,10 +95,10 @@ class MpHistoryItemManager(val historyItemDao: HistoryItemEntityDao): HistoryIte
         for(report in reports) {
             val dateBucket = getDateBucket(report)
             val details = createMeasurementHistoryDetails(type, report)
-            val historyItem = HistoryItem(type, report.identifier!!, dateBucket, report.dateTime!!, details)
+            val historyItem = HistoryItem(type, report.identifier?:"", dateBucket, report.dateTime?: Instant.now(), details)
```

Comment:
I am worried that by preventing a crash here, we could be presenting users with incorrect data.  By leaving the !! we would trigger a crash that would likely be found during QA. There may be a better fix that avoids displaying wrong data, but for now let's go back to !! instead of Instant.now() in this file.

Reply:
Did you mean to return back to !! only for this function or for all functions in the file? And only dateTime or other file(like identifier) also?

Reply:
All functions in the file with the added Instant.now().  I'm less worried about the other fields.

Reply:
got you. I will fix it. Thank you.